### ISSUE TITLE
In-Game: [Ubuntu] buffer_load() on a file always just returns -1

### ISSUE THREAD

**Issue by patchuby** [YoYoGames/GameMaker-Bugs#4644]
### Description
I'm trying to load save files using buffer_load. It works fine on Windows and Mac but fails here and returns -1

### Steps To Reproduce
1. Launch the game
2. Start a new game
3. After the end of the scene (when you can move around), exit via escape to the main menu (this step is only to create a sav)
4. Look at the output, you should see error messages from failing to read -1 as a buffer and the file that tried to load

The code is on line 14 of the script "DataStructures"

[36ceec0c-4db6-4808-875d-160bb796ef4a](https://api.gamemaker.io/api/github/downloads/36ceec0c-4db6-4808-875d-160bb796ef4a)

Submission attempt 2:
[cc4897f2-2395-406f-962b-e0bef95ab88d](https://api.gamemaker.io/api/github/downloads/cc4897f2-2395-406f-962b-e0bef95ab88d)

**Comment by patchuby** [comment]
@yoyo-danielc Providing the full path (game_save_id + file name) instead of the relative path seems to solve the issue.

**Comment by yoyo-danielc** [comment]
Hi @patchuby,

Could you provide a more minimal sample? The project attached to this crashes for me after the second splash screen on both Windows and Linux under 2023.11 and the latest 2024.02 beta.

I had a glance at your code, and I _believe_ it would simply be doing `buffer_save()` and `buffer_load()` using `game_save_id + "foo"` as a path, which seems to work fine for me on both Windows and Linux:

```
var b = buffer_create(1, buffer_grow, 1);
buffer_write(b, buffer_text, "Hello world");
buffer_save(b, "file1.txt");
buffer_save(b, game_save_id + "file2.txt");
buffer_delete(b);

b = buffer_load("file1.txt");
show_debug_message("Loaded file1.txt: " + buffer_read(b, buffer_text));
buffer_delete(b);

b = buffer_load(game_save_id + "file2.txt");
show_debug_message("Loaded file2.txt: " + buffer_read(b, buffer_text));
buffer_delete(b);
```

**Comment by patchuby** [comment]
@yoyo-danielc So relative paths don't work for buffer_load ? It works on Windows for me.
I'm sorry I don't have time to isolate the issue right now.
Also don't know why it crashes for you, is there a crash message ?
I tested it from a Windows IDE running the ubuntu run to a separate laptop, don't know if that could affect it. It definitely runs fine on Windows in any case. I'm on 2023.11.1.160.

Since using game_save_id works for now, this can definitely go back on the 2024.4 milestone as far as I'm concerned , and is much less crucial than 
https://github.com/YoYoGames/GameMaker-Bugs/issues/4652

**Comment by yoyo-danielc** [comment]
I mean, it works using a relative path as well as an absolute one for me. This is the error I get when the project crashes:

```
***********************************************************TEST DES ROOMS POUR LE CHAPITRE 0********************************************************************
ERROR!!! :: ############################################################################################
ERROR in
action number 1
of  Step Event0
for object obj_god:


Variable <unknown_object>.width(100120, -2147483648) cannot be resolved.
 at gml_Script_test_maps (line 29) - 		var cols = map_data.width * navgrid_subdivisions,
############################################################################################
gml_Script_test_maps (line 29)
gml_Script_execute_tests (line 34) - 	test_maps(global.chapter_maps_raw);
gml_Script_load_chapter_data_2 (line 240) - 	if(debug_mode_enabled) execute_tests();
gml_Object_obj_god_Step_0 (line 108)
```

**Comment by patchuby** [comment]
> I mean, it works using a relative path as well as an absolute one for me. This is the error I get when the project crashes:
> 
> ```
> ***********************************************************TEST DES ROOMS POUR LE CHAPITRE 0********************************************************************
> ERROR!!! :: ############################################################################################
> ERROR in
> action number 1
> of  Step Event0
> for object obj_god:
> 
> 
> Variable <unknown_object>.width(100120, -2147483648) cannot be resolved.
>  at gml_Script_test_maps (line 29) - 		var cols = map_data.width * navgrid_subdivisions,
> ############################################################################################
> gml_Script_test_maps (line 29)
> gml_Script_execute_tests (line 34) - 	test_maps(global.chapter_maps_raw);
> gml_Script_load_chapter_data_2 (line 240) - 	if(debug_mode_enabled) execute_tests();
> gml_Object_obj_god_Step_0 (line 108)
> ```

Ah yes this was from a another issue caused by GM forcing lower case on all file names on Linux & Mac. It should be fixed now. How can I resubmit the project ?

**Comment by yoyo-danielc** [comment]
> Ah yes this was from a another issue caused by GM forcing lower case on all file names on Linux & Mac. It should be fixed now. How can I resubmit the project ?

I don't think you can replace the project in the original submission, you would have to resubmit a new issue from the IDE.

**Comment by patchuby** [comment]
@yoyo-danielc I've resubmitted it privately (not to clutter the public tracker), it's the issue 248

**Comment by YYDan** [comment]
I have now added the link to the description here.
